src/main/java/io/github/ibuildthecloud/gdapi/validation/ValidationHandler.java

```diff
@@ -338,11 +345,10 @@ protected Object checkType(String fieldName, Object value, Class<?> type) {
             result.addAll(items);
             return result;
         }
-
         FieldType type = subTypes.get(0);
         for ( Object item : items ) {
-            item = convert(fieldName, null, type, subTypes.subList(1, subTypes.size()),
-                    subTypesNames.subList(1, subTypesNames.size()), item, context);
+            item = convert(fieldName, null, type, subTypes.subList(0, subTypes.size()),
```

Comment:
If we are only supporting a single nest, I can change the code here to simply send the list down.  No need for sublist.